src/main/java/com/thealgorithms/datastructures/graphs/HamiltonianCycle.java

```diff
@@ -0,0 +1,127 @@
+package com.thealgorithms.datastructures.graphs;
+
+/**
+ * Java program for Hamiltonian Cycle (https://en.wikipedia.org/wiki/Hamiltonian_path).
+ * @author Akshay Dubey (https://github.com/itsAkshayDubey)
+ */
+public class HamiltonianCycle {
+
+	private int V, pathCount;
+	private int[] path;
+	private int[][] graph;
+
+	/**
+	 * Find hamiltonian cycle for given graph G(V,E)
+	 *
+	 * @param graph Adjacency matrix of a graph G(V, E) 
+	 * for which hamiltonian path is to be found
+	 * @return Array containing hamiltonian cycle else returns 1D array with value -1.
+	 */
+	public int[] findHamiltonianCycle(int[][] graph){
+
+		this.V = graph.length;
+		this.path = new int[this.V];
+
+		//Initialize path array with -1 value
+		for(int i=0 ; i<this.path.length ; i++) {
+			this.path[i] = -1;
+		}
+
+		this.graph = graph;        
+
+		try{            
+			this.path[0] = 0;
+			this.pathCount = 1;            
+
+			solve(0);
+
+			for(int i=0 ; i<this.path.length ; i++) {
+				this.path[i] = -1;
+			}
+
+		}
+
+		catch (Exception e){
+
+		}
+		return path;
+	}
+
+	/** function to find paths recursively **/
+	/**
+	 * Find paths recursively from given vertex
+	 *
+	 * @param vertex Vertex from which path is to be found
+	 */
+	public void solve(int vertex) throws Exception{
+
+		if (this.graph[vertex][0] == 1 && this.pathCount == this.V) {
+
+			throw new Exception("Solution found");
```

Comment:
Please don't use exceptions for flow control. return found path from the function instead

Reply:
Hello @siriak 

Formatted code in commit: 57d46aa5c1896209afe45875bd2670b33ba1d1dd, fixed flow of solve method and renamed + changed return type of _void solve(int x)_ to  _boolean isPathFound(int x)_ in commit: 7c343a92fc21921561a6b6f4cdcd4af5ed5daeb5

Kindly have a look.

Thanks.